src/amqp_frame_reader.ts

```diff
@@ -1,24 +1,25 @@
 import { FrameError } from "./frame_error.ts";
 import { decodeHeader, decodeMethod } from "./amqp_codec.ts";
 import type { IncomingFrame } from "./amqp_frame.ts";
-import { BufReader } from "../deps.ts";
+import type { Reader } from "jsr:@std/io/types";
 
 export class AmqpFrameReader {
   #timer: null | number = null;
-  #reader: BufReader;
+  #reader: Reader;
 
-  constructor(r: Deno.Reader) {
-    this.#reader = BufReader.create(r);
+  constructor(r: Reader) {
+    this.#reader = r;
   }
 
   #readBytes = async (length: number): Promise<Uint8Array> => {
-    const n = await this.#reader.readFull(new Uint8Array(length));
+    const buf = new Uint8Array(length);
+    const n = await this.#reader.read(buf);
 
     if (n === null) {
       throw new FrameError("EOF");
     }
 
-    return n;
+    return buf;
```

Comment:
I am wondering if this might cause issues on slower connections. From the docs https://docs.deno.com/api/deno/~/Deno.Conn#call_signature_read_0:

> It is possible for a read to successfully return with 0 bytes. This does not indicate EOF.
> It is not guaranteed that the full buffer will be read in a single call.

These are the cases that BufReader handled. I see the tests passes though. But I do not remember if there is a test that specifically triggers the case where the full buffer is not read in a single call.